[CCI] Index out of range panic

I got a panic when I search for a CCI namespace and it does not have network.
```
panic: runtime error: index out of range [0] with length 0

goroutine 236 [running]:
github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/cci.flattenVpcNetwork(...)
        /home/lance/go/src/github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/cci/data_source_huaweicloud_cci_namespaces.go:165
github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/cci.filterDataCciNamespaces(0xc001563c00, 0xc00146f590, 0xc000f41180, 0x1, 0x1, 0xc000f41180, 0x0, 0x1, 0x0, 0xc000cbfce0, ...)
        /home/lance/go/src/github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/cci/data_source_huaweicloud_cci_namespaces.go:204 +0x1199
github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/cci.dataSourceCciNamespacesRead(0x1f42308, 0xc000ca9f20, 0xc001563c00, 0x1c9c580, 0xc000956400, 0xc000f24ce0, 0xc0006fd8b8, 0x410178)
        /home/lance/go/src/github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/cci/data_source_huaweicloud_cci_namespaces.go:245 +0x62b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc000214380, 0x1f42298, 0xc0011628c0, 0xc001563c00, 0x1c9c580, 0xc000956400, 0x0, 0x0, 0x0)
```

--- a/huaweicloud/services/acceptance/cci/data_source_huaweicloud_cci_namespaces_test.go
+++ b/huaweicloud/services/acceptance/cci/data_source_huaweicloud_cci_namespaces_test.go
@@ -46,6 +46,26 @@ func TestAccDataCciNamespaces_basic(t *testing.T) {
 	})
 }
 
+func TestAccDataCciNamespaces_noNetwork(t *testing.T) {
+	dataSourceName := "data.huaweicloud_cci_namespaces.test"
+	rName := acceptance.RandomAccResourceNameWithDash()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataCciNamespaces_noNetwork(rName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(dataSourceName, "namespaces.#", "1"),
+					resource.TestCheckResourceAttr(dataSourceName, "namespaces.0.name", rName),
+					resource.TestCheckResourceAttr(dataSourceName, "namespaces.0.type", "general-computing"),
+				),
+			},
+		},
+	})
+}
+
 func testAccDataCciNamespaces_base(rName string) string {
 	return fmt.Sprintf(`
 data "huaweicloud_availability_zones" "test" {}
@@ -92,4 +112,17 @@ data "huaweicloud_cci_namespaces" "test" {
   type = "general-computing"
 }
 `, testAccDataCciNamespaces_base(rName), rName)
+}
+
+func testAccDataCciNamespaces_noNetwork(rName string) string {
+	return fmt.Sprintf(`
+resource "huaweicloud_cci_namespace" "test" {
+  name = "%[1]s"
+  type = "general-computing"
+}
+
+data "huaweicloud_cci_namespaces" "test" {
+  name = huaweicloud_cci_namespace.test.name
+}
+`, rName)
 }

--- a/huaweicloud/services/cci/data_source_huaweicloud_cci_namespaces.go
+++ b/huaweicloud/services/cci/data_source_huaweicloud_cci_namespaces.go
@@ -157,7 +157,12 @@ func isNamespaceParamMatch(d *schema.ResourceData, ns namespaces.Namespace) bool
 	return true
 }
 
-func flattenVpcNetwork(network networks.Network) []map[string]interface{} {
+func flattenVpcNetwork(networkList []networks.Network) []map[string]interface{} {
+	if len(networkList) < 1 {
+		return nil
+	}
+
+	network := networkList[0]
 	return []map[string]interface{}{
 		{
 			"name":              network.Metadata.Name,
@@ -196,7 +201,7 @@ func filterDataCciNamespaces(d *schema.ResourceData, client *golangsdk.ServiceCl
 				"rbac_enabled":              ns.Metadata.Labels.RbacEnable,
 				"created_at":                ns.Metadata.CreationTimestamp,
 				"status":                    ns.Status.Phase,
-				"network":                   flattenVpcNetwork(netList[0]),
+				"network":                   flattenVpcNetwork(netList),
 			}
 
 			result = append(result, nsParams)

